### PR TITLE
[pytest] Add test_gen_spytestbed.py to generate Spytest testbed file

### PR DESCRIPTION
**Pull request by lolyu**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

* Add template file `tests/template/spytest_testbed.yaml.j2` as the
testbed template.
* Add a testcase:
`tests/testbed_setup/test_gen_spy_testbed.py::test_gen_spy_testbed` to
generate the testbed file for Spytest and save as
`spytest/testbeds/spytest_testbed.yaml`.

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
To automate the process to generate testbed file from the connection graph files.

#### How did you verify/test it?
```
pytest --inventory ../ansible/lab --host-pattern all -vvv --show-capture stdout --testbed spytest --testbed_file ../ansible/testbed.csv --disable_loganalyzer --log-file-level debug --pdb --skip_sanity -s testbed_setup/test_gen_spy_testbed.py
```

### CODE REVIEW

**Comment by lolyu** [comment]
retest this please.

File: tests/testbed_setup/test_gen_spy_testbed.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    intf = 1
+    for hostname in hostnames:
+        end_device = hostname
+        conns = connections[hostname]
+        end_ports = sorted(
+            (_ for _ in conns if conns[_]['peerdevice'] not in connections),
+            key=_interface_key)
+        for end_port in end_ports:
+            ptf_conn = {
+                "start_port": PTF_INTERFACE_TEMPLATE % intf,
```
Comment (daall) [inline tests/testbed_setup/test_gen_spy_testbed.py:70]: This requires the DUTs to be connected to the PTF sequentially, right? Do we have that guarantee in the testbed file?

Comment (lolyu) [reply tests/testbed_setup/test_gen_spy_testbed.py:70]: Yes, the interfaces in PTF are inserted sequentially in the same order as the source DUTs defined in `testbed.csv`.
For instance, the `testbed.csv` contains an entry like:
```
spytest,spytest,fullmesh,docker-ptf-brcm,ptf-unknown,192.168.0.1/24,,server_16,,[lab-msn2700-spy-1;lab-msn2700-spy-2;lab-a7060-spy-1;lab-a7060-spy-2],hellospytest
```
And the `device_vlan_list` for duts returned by `conn_graph_facts` will be like `[[190, 191, 192, 193], [202, 203, 204, 205], [206, 207, 208, 209], [210, 211, 212, 213]]`. The interfaces will be `Eth0` ~ `Ether15`, which are in the same order as flattened `device_vlan_list`.

Comment (daall) [reply tests/testbed_setup/test_gen_spy_testbed.py:70]: Cool, thank you for the explanation! :)

**Pull request merged**